src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -0,0 +1,526 @@
+    if(flags & PhongMaterialData::Flag::AmbientTexture)
+        data->ambientColor() = Color3(color);
+
+    mat->Get(AI_MATKEY_COLOR_DIFFUSE, color); /* Key always present, default black */
+    if(flags & PhongMaterialData::Flag::DiffuseTexture)
+        data->diffuseColor() = Color3(color);
+
+    mat->Get(AI_MATKEY_COLOR_SPECULAR, color); /* Key always present, default black */
+    if(flags & PhongMaterialData::Flag::SpecularTexture)
+        data->specularColor() = Color3(color);
+
+    return std::move(data);
+}
+
+UnsignedInt AssimpImporter::doTextureCount() const { return _textures.size(); }
+
+std::optional<TextureData> AssimpImporter::doTexture(const UnsignedInt id) {
+    constexpr Sampler::Filter filterU = Sampler::Filter::Linear;
+    constexpr Sampler::Filter filterV = Sampler::Filter::Linear;
+
+    /* Load texture wrapping property */
+    Sampler::Wrapping wrappingU = Sampler::Wrapping::ClampToEdge;
+    Sampler::Wrapping wrappingV = Sampler::Wrapping::ClampToEdge;
+    constexpr Sampler::Wrapping wrappingW = Sampler::Wrapping::ClampToEdge;
+
+    auto toWrapping = [](aiTextureMapMode mapMode) {
+        switch (mapMode) {
+        case aiTextureMapMode_Wrap:
+            return Sampler::Wrapping::Repeat;
+        case aiTextureMapMode_Decal:
+            Warning() << "Trade::AssimpImporter::texture(): no wrapping enum to match aiTextureMapMode_Decal, using Sampler::Wrapping::ClampToEdge.";
+        case aiTextureMapMode_Clamp:
+            return Sampler::Wrapping::ClampToEdge;
+        case aiTextureMapMode_Mirror:
+            return Sampler::Wrapping::MirroredRepeat;
+        }
+    };
+
+    aiTextureMapMode mapMode;
+    const aiMaterial* mat = _textures[id].first;
+    const aiTextureType type = _textures[id].second;
+    if(mat->Get(AI_MATKEY_MAPPINGMODE_U(type, 0), mapMode) == AI_SUCCESS) {
+        wrappingU = toWrapping(mapMode);
+    }
+    if(mat->Get(AI_MATKEY_MAPPINGMODE_V(type, 0), mapMode) == AI_SUCCESS) {
+        wrappingV = toWrapping(mapMode);
+    }
+
+    return TextureData{TextureData::Type::Texture2D, filterU, filterV, {Sampler::Mipmap::Linear}, {wrappingU, wrappingV, wrappingW}, id, &_textures[id]};
+}
+
+UnsignedInt AssimpImporter::doImage2DCount() const { return _textures.size(); }
+
+std::optional<ImageData2D> AssimpImporter::doImage2D(const UnsignedInt id) {
+    const aiMaterial* mat;
+    aiTextureType type;
+    std::tie(mat, type) = _textures[id];
+
+    aiString texturePath;
+    if(mat->Get(AI_MATKEY_TEXTURE(type, 0), texturePath) != AI_SUCCESS) {
+        Error() << "Trade::AssimpImporter::image2D(): Error getting path for texture with id" << id;
+        return std::nullopt;
+    }
+
+    std::string path = texturePath.C_Str();
+    if(path[0] == '*') {
+        /* Load embedded texture */
+        char* err;
+        const Int index = Int(strtol(path.substr(0).c_str(), &err, 10));
+        if(err == nullptr) {
```

Comment:
Yes, it's the pointer to the first char after the number. If no number, no position after, hence nullptr.
Maybe the naming of my var is confusing there